File: exercise-3/starter/src/finding.ts
import type { SDK } from "caido:workflow";

import type { Finding } from "./analyse";

export async function createFinding(sdk: SDK, finding: Finding) {
  let description = "Found reflected parameters in reponse:\n";
  for (let [key, value] of finding.parameters) {
    description += `- ${key}: ${value}\n`;
  }

  const result = {}; // CODE: Create a finding

  // @ts-ignore
  sdk.console.log(`Finding created with ID ${result.getId()}`);
}

File: exercise-7/starter/src/finding.py
from gql.client import AsyncClientSession

from .gql import gql
from .analyze import Finding


async def create_finding(client: AsyncClientSession, finding: Finding):
    query = gql(
        """
        mutation CreateFinding($requestId: ID!, $input: CreateFindingInput!) {
            # CODE: Mutation to create a finding with requestId and input
        }
        """
    )
    description = "Found reflected parameters in reponse:\n"
    for key, value in finding.parameters:
        description += f"- {key}: {value}\n"

    finding = await client.execute(
        query,
        variable_values={
            "requestId": finding.id,
            "input": {
                "dedupeKey": "",  # CODE: Set the dedupeKey to a unique value,
                "description": "",  # CODE: Set the description to the description variable,
                "reporter": "Reflector",
                "title": "Reflected parameters",
            },
        },
    )

    if finding["createFinding"]["finding"]:
        print(f"[-] Created finding {finding['createFinding']['finding']['id']}")
